lib/pos_leader_schedule.go

```diff
@@ -19,10 +19,14 @@ func (bav *UtxoView) GenerateLeaderSchedule() ([]*PKID, error) {
 	// to snapshot the leader schedule.
 	currentGlobalParamsEntry := bav.GetCurrentGlobalParamsEntry()
 
-	// Retrieve top, active validators ordered by stake.
-	validatorEntries, err := bav.GetTopActiveValidatorsByStakeAmount(currentGlobalParamsEntry.LeaderScheduleMaxNumValidators)
-	if err != nil {
-		return nil, errors.Wrapf(err, "UtxoView.GenerateLeaderSchedule: error retrieving top ValidatorEntries: ")
+	var validatorEntries []*ValidatorEntry
+	// If the number of validators is less than the max number of validators, we can use all of them.
+	if uint64(len(maxValidatorSet)) <= currentGlobalParamsEntry.LeaderScheduleMaxNumValidators {
+		validatorEntries = maxValidatorSet[:]
+	} else {
```

Comment:
No need to create a copy of the validatorSet slice
```suggestion
	validatorEntries := validatorSet
	if uint64(len(maxValidatorSet)) > currentGlobalParamsEntry.LeaderScheduleMaxNumValidators {
```